Couldn't get the value "DistributionName" on Windows 11

```powershell
PS C:\Users\echo\Desktop> lxrunoffline.exe install `
>> -n AOSC `
>> -d C:\files\AOSC\ `
>> -f C:\Users\echo\Downloads\aosc-os_base+nvidia_20210602_amd64.tar.xz `
>> -s
[ERROR] Couldn't get the value "DistributionName" of the registry key "Software\Microsoft\Windows\CurrentVersion\Lxss\AppxInstallerCache".
Reason: 系统找不到指定的文件。
// Reason: System could not found specific file.
```

--- a/src/lib/include/LxRunOffline/pch.h
+++ b/src/lib/include/LxRunOffline/pch.h
@@ -26,6 +26,7 @@
 #include <map>
 #include <memory>
 #include <numeric>
+#include <regex>
 #include <set>
 #include <sstream>
 #include <stack>

--- a/src/lib/reg.cpp
+++ b/src/lib/reg.cpp
@@ -34,7 +34,9 @@ void fclose_safe(FILE *f) {
 }
 
 static bool is_guid(crwstr str) {
-	static const std::wregex guid_regex(LR"#(\{[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}\})#");
+	static const std::wregex guid_regex(
+		LR"#(\{[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}\})#",
+		std::regex_constants::icase);
 	return std::regex_match(str, guid_regex);
 }
 

--- a/src/lib/reg.cpp
+++ b/src/lib/reg.cpp
@@ -33,6 +33,11 @@ void fclose_safe(FILE *f) {
 	if (f) fclose(f);
 }
 
+static bool is_guid(wstr str) {
+	std::wregex guid_regex(R"#(\{[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}\})#");
+	return std::regex_match(str, guid_regex);
+}
+
 static wstr new_guid() {
 	GUID guid;
 	const auto hr = CoCreateGuid(&guid);
@@ -144,7 +149,7 @@ static std::vector<wstr> list_distro_id() {
 		const auto code = RegEnumKeyEx(hk.get(), i, ib.get(), &bs, nullptr, nullptr, nullptr, nullptr);
 		if (code == ERROR_NO_MORE_ITEMS) break;
 		else if (code) throw lro_error::from_win32(err_msg::err_enum_key, { reg_base_path }, code);
-		res.emplace_back(ib.get());
+		if (is_guid(ib.get())) res.emplace_back(ib.get());
 	}
 	return res;
 }

--- a/src/lib/reg.cpp
+++ b/src/lib/reg.cpp
@@ -33,8 +33,8 @@ void fclose_safe(FILE *f) {
 	if (f) fclose(f);
 }
 
-static bool is_guid(wstr str) {
-	std::wregex guid_regex(R"#(\{[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}\})#");
+static bool is_guid(crwstr str) {
+	static const std::wregex guid_regex(LR"#(\{[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}\})#");
 	return std::regex_match(str, guid_regex);
 }
 
@@ -148,6 +148,7 @@ static std::vector<wstr> list_distro_id() {
 		DWORD bs = guid_len + 1;
 		const auto code = RegEnumKeyEx(hk.get(), i, ib.get(), &bs, nullptr, nullptr, nullptr, nullptr);
 		if (code == ERROR_NO_MORE_ITEMS) break;
+		else if (code == ERROR_MORE_DATA) continue;
 		else if (code) throw lro_error::from_win32(err_msg::err_enum_key, { reg_base_path }, code);
 		if (is_guid(ib.get())) res.emplace_back(ib.get());
 	}

--- a/tests/test_reg.cpp
+++ b/tests/test_reg.cpp
@@ -133,6 +133,9 @@ BOOST_FIXTURE_TEST_CASE(test_list_distros, fixture_tmp_reg) {
 	register_test_distro(root_key, L"foo1", L"C:\\bar1");
 	register_test_distro(root_key, L"foo2", L"C:\\bar2");
 	register_test_distro(root_key, L"foo3", L"C:\\bar3");
+	const auto non_guid_key = root_key.open_subkey(L"AppxInstallerCache", true);
+	const auto long_key = root_key.open_subkey(
+		L"a string which is longer than the buffer for GUID in list_distro_id", true);
 	auto list = list_distros();
 	std::sort(list.begin(), list.end());
 	BOOST_TEST(list == (std::vector<std::wstring> { L"foo1", L"foo2", L"foo3" }));

